src/dowel/csv_output.py

```diff
@@ -1,79 +1,119 @@
-"""A `dowel.logger.LogOutput` for CSV files."""
-import csv
-import warnings
-
-from dowel import TabularInput
-from dowel.simple_outputs import FileOutput
-from dowel.utils import colorize
-
-
-class CsvOutput(FileOutput):
-    """CSV file output for logger.
-
-    :param file_name: The file this output should log to.
-    """
-
-    def __init__(self, file_name):
-        super().__init__(file_name)
-        self._writer = None
-        self._fieldnames = None
-        self._warned_once = set()
-        self._disable_warnings = False
-
-    @property
-    def types_accepted(self):
-        """Accept TabularInput objects only."""
-        return (TabularInput, )
-
-    def record(self, data, prefix=''):
-        """Log tabular data to CSV."""
-        if isinstance(data, TabularInput):
-            to_csv = data.as_primitive_dict
-
-            if not to_csv.keys() and not self._writer:
-                return
-
-            if not self._writer:
-                self._fieldnames = set(to_csv.keys())
-                self._writer = csv.DictWriter(
-                    self._log_file,
-                    fieldnames=self._fieldnames,
-                    extrasaction='ignore')
-                self._writer.writeheader()
-
-            if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
```

Comment:
Now that the call to `_warn` has been deleted, is there any other code that can also be deleted?

Reply:
removed code involving warnings in csv_output.py and test_csv_output.py